pkgs/build-support/setup-hooks/make-symlinks-relative.sh

```diff
@@ -0,0 +1,23 @@
+fixupOutputHooks+=(_makeSymlinksRelative)
+
+# For every symlink in $output that refers to another file in $output
+# ensure that the symlink is relative. This removes references to the output
+# has from the resulting store paths and thus the NAR files.
+_makeSymlinksRelative() {
+    local symlinkTarget
+    local relativeTarget
+    while IFS= read -r -d $'\0' f; do
+        symlinkTarget=$(readlink "$f")
+        if [ "${symlinkTarget:0:${#prefix}}" != "$prefix" ]; then
+            # skip this symlink as it doesn't point to $prefix
+            continue
+        fi
+
+        relativeTarget="${symlinkTarget:${#prefix}}"
+        echo "rewriting symlink $f to be relative to $prefix"
+
+        rm "$f"
+        ln -rs "$prefix/$relativeTarget" "$f"
```

Comment:
The `ln` destination should be `"$prefix/$f"`.

To simplify the process this block can be replaced with these `ln` flags:
```bash
        echo "rewriting symlink $f to be relative to $prefix"
        ln -snrf "$symlinkTarget" "$prefix/$f"
```
as `ln -r` produces the minimal possible relative link.